src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CaseMCacheUpdater.java

```diff
@@ -0,0 +1,839 @@
+      } else {
+        kuntaApiParentId = translatePageId(toNodeId(organizationId, parentNode.getNodeId()), false);
+      }
+    }
+    
+    List<LocalizedValue> titles = translateNodeNames(organizationId, node.getNames());
+    page.setId(kuntaApiId.getId());
+    page.setParentId(kuntaApiParentId != null ? kuntaApiParentId.getId() : null);
+    page.setTitles(titles);
+    page.setSlug(slugify(titles.isEmpty() ? kuntaApiId.getId() : titles.get(0).getValue()));
+    
+    return page;
+  }
+  
+  private Page translateContent(OrganizationId organizationId, Page parentPage, Content content, String title, String slug) {
+    Page page = new Page();
+    
+    page.setId(translatePageId(toContentId(organizationId, content.getContentId()), true).getId());
+    page.setParentId(parentPage.getId());
+    page.setTitles(toTitles(title));
+    page.setSlug(slug);
+
+    return page;
+  }
+  
+  private PageId translatePageId(PageId pageId, boolean createMissing) {
+    PageId result = discoveredPageIds.get(pageId);
+    if (result != null) {
+      return result;
+    }
+    
+    result = idController.translatePageId(pageId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (result != null) {
+      return result;
+    }
+    
+    if (createMissing) {
+      Identifier identifier = identifierController.createIdentifier(pageId);
+      result = new PageId(KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+      discoveredPageIds.put(pageId, result);
+    }
+    
+    return result;
+  }
+  
+  private PageId toNodeId(OrganizationId organizationId, Long nodeId) {
+    return new PageId(CaseMConsts.IDENTIFIER_NAME, String.format("%s|NODE|%d", organizationId.getId(), nodeId));
+  }
+  
+  private PageId toContentId(OrganizationId organizationId, Long contentId) {
+    return new PageId(CaseMConsts.IDENTIFIER_NAME, String.format("%s|CONTENT|%d", organizationId.getId(), contentId));
+  }
+  
+
+  private String slugify(String title) {
+    return slugify(title, MAX_SLUG_LENGTH );
+  }
+  
+  private String slugify(String text, int maxLength) {
+    String urlName = StringUtils.normalizeSpace(text);
+    if (StringUtils.isBlank(urlName))
+      return UUID.randomUUID().toString();
+    
+    urlName = StringUtils.lowerCase(StringUtils.substring(StringUtils.stripAccents(urlName.replaceAll(" ", "_")).replaceAll("[^a-zA-Z0-9\\-\\.\\_]", ""), 0, maxLength));
+    if (StringUtils.isBlank(urlName)) {
+      urlName = UUID.randomUUID().toString();
+    }
+    
+    return urlName;
+  }
+  
+  private List<LocalizedValue> toTitles(String title) {
+    LocalizedValue localizedValue = new LocalizedValue();
+    localizedValue.setLanguage(CaseMConsts.DEFAULT_LANGUAGE);
+    localizedValue.setValue(title);
+    return Collections.singletonList(localizedValue);
+  }
+
+  private List<LocalizedValue> translateNodeNames(OrganizationId organizationId, List<NodeName> names) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateNodeNames" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)